egs/wsj/s5/utils/subword/prepare_lang_subword.py

```diff
@@ -353,34 +338,8 @@
   perl -ane '@A = split; print $A[0], " ", join(" ", @A), "\n";' | sort | uniq > $dir/phones/align_lexicon.txt
 
 if [ -f $srcdir/nonterminals.txt ]; then
-#  utils/lang/grammar/augment_phones_txt.py $dir/phones.txt $srcdir/nonterminals.txt $dir/phones.txt
-#  utils/lang/grammar/augment_words_txt.py $dir/words.txt $srcdir/nonterminals.txt $dir/words.txt
-#  cp $srcdir/nonterminals.txt $dir/phones/nonterminals.txt
-#  utils/sym2int.pl $dir/phones.txt <$dir/phones/nonterminals.txt >$dir/phones/nonterminals.int
-#
-#  for w in "#nonterm_begin" "#nonterm_end" $(cat $srcdir/nonterminals.txt); do
-#    echo $w $w  # These are words without pronunciations, so leave those prons
-#                # empty.
-#  done >> $dir/phones/align_lexicon.txt
-#  nonterm_phones_offset=$(grep '#nonterm_bos' <$dir/phones.txt | awk '{print $2}')
-#  echo $nonterm_phones_offset > $dir/phones/nonterm_phones_offset.int
-#  echo '#nonterm_bos' > $dir/phones/nonterm_phones_offset.txt  # temporary.
-#
-#  if [ -f $dir/phones/word_boundary.txt ]; then
-#    # word-position-dependent system.  Only include the optional-silence phone,
-#    # and phones that can end a word, plus the special symbol #nonterm_bos, in the
-#    # left-context phones.
-#    awk '{if ($2 == "end" || $2 == "singleton") print $1; }' <$dir/phones/word_boundary.txt | \
-#        cat - $dir/phones/optional_silence.txt $dir/phones/nonterm_phones_offset.txt > $dir/phones/left_context_phones.txt
-#  else
-#    cat $dir/phones/{silence,nonsilence}.txt $dir/phones/nonterm_phones_offset.txt > $dir/phones/left_context_phones.txt
-#  fi
-#  utils/sym2int.pl $dir/phones.txt <$dir/phones/left_context_phones.txt >$dir/phones/left_context_phones.int
-#
-#  # we need to write utils/lang/make_lexicon_fst_silprob.py before this can work.
-#  grammar_opts="--left-context-phones=$dir/phones/left_context_phones.txt --nonterminals=$srcdir/nonterminals.txt"
-#else
-  echo "Currently do not support nonterminals"
+  echo "$0: Currently we do not support nonterminals" && exit 1;
```

Comment:
Instead of checking conditions here, it's better if you check all conditions in the beginning of the script, so if it fails it happens immediately.